workflow/rules/qc.smk

```diff
@@ -0,0 +1,130 @@
+rule snp_filters_qc:
+    input:
+        vcf = config['output'] + "{Organism}/{refGenome}/" + "{Organism}_{refGenome}.final.vcf.gz"
+    output:
+        snpqc = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}_snpqc.txt"
+    conda:
+        "../envs/qc.yml"
+    params:
+        prefix=config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}"
+    shell:
+        "bcftools query -f '%CHROM\t%POS\t%ID\t%INFO/AF\t%QUAL\t%INFO/ReadPosRankSum\t%INFO/FS\t%INFO/SOR\t%INFO/MQ\t%INFO/MQRankSum\n' {input.vcf} > {output.snpqc}"
+
+rule vcftools_individuals:
+    input:
+        vcf = config['output'] + "{Organism}/{refGenome}/" + "{Organism}_{refGenome}.final.vcf.gz"
+        
+    output:
+        depth = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}.idepth",
+        miss = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}.imiss"
+    conda:
+        "../envs/qc.yml"
+    params:
+        prefix=config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}"
+    shell:
+        """
+        vcftools --gzvcf {input.vcf} --FILTER-summary --out {params.prefix}
+        vcftools --gzvcf {input.vcf} --out {params.prefix} --depth
+        vcftools --gzvcf {input.vcf} --out {params.prefix} --missing-indv
+        """
+
+rule subsample_snps:
+    input:
+        vcf = config['output'] + "{Organism}/{refGenome}/" + "{Organism}_{refGenome}.final.vcf.gz"
+    output:
+        filt_snps = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}_filt_snps.txt",
+        tmp_rand_snps = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}_tmp_rand_snps.txt",
+        filtered = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}.filtered.vcf.gz",
+        filtered_idx = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}.filtered.vcf.gz.csi",
+        pruned = config['output'] + "{Organism}/{refGenome}/" + config['qcDir'] + "{Organism}_{refGenome}.pruned.vcf.gz"
+    conda:
+        "../envs/qc.yml"
+    resources:
+        mem_mb = lambda wildcards, attempt: attempt * res_config['vcftools']['mem']    # this is the overall memory requested
+    shell:
+        """
+        bcftools view -v snps -m2 -M2 -f .,PASS -e 'AF==1 | AF==0 | ALT="*" | TYPE~"indel" | ref="N"' {input.vcf} -O z -o {output.filtered}
+        bcftools index {output.filtered}
+        bcftools query -f '%CHROM\t%POS\n' {output.filtered} > {output.filt_snps}
+        sort -R {output.filt_snps}| head -50000 > {output.tmp_rand_snps}
```

Comment:
I agree, definitely an issue if one wants to e.g. interpret the admixture plot beyond QC. I was thinking of speed / QC with this default, but we can try with a default r2 as you suggest and see how it goes

Reply:
Ideally I guess this would be a user-settable parameter, basically have a 'quick QC' option and a 'full pass' option or something. For now let's see how much longer it takes with just some LD pruning. 

Maybe we could start with having this be a config option with two choices, 'fast-pruned' or 'LD-pruned' or similar, and have the fast-pruned option implement the current code, and have the LD option implement a r2 filter only but not otherwise downsample. Then we can test on various datasets and decide how to proceed? That should be pretty easy to code up.

Reply:
I pushed a commit for this. What I decided to do is count how many variants are in the filtered VCF file, then calculate a SNP-based window to get to ~100k SNPs. This seems like a reasonably safe way to have a spatial component to getting a random subset of SNPs that is fast and won't break easily.